server/src/main/java/org/opensearch/index/codec/CodecService.java

```diff
@@ -58,29 +61,64 @@ public class CodecService {
 
     public static final String DEFAULT_CODEC = "default";
     public static final String BEST_COMPRESSION_CODEC = "best_compression";
-    /** the raw unfiltered lucene default. useful for testing */
+    /**
+     * the raw unfiltered lucene default. useful for testing
+     */
     public static final String LUCENE_DEFAULT_CODEC = "lucene_default";
     public static final String ZSTD_CODEC = "zstd";
     public static final String ZSTD_NO_DICT_CODEC = "zstd_no_dict";
 
+    public CodecService(@Nullable MapperService mapperService, IndexSettings indexSettings, Logger logger) {
+        final MapBuilder<String, Codec> codecs = MapBuilder.<String, Codec>newMapBuilder();
+        assert null != indexSettings;
+        addIndexCodecs(codecs, mapperService, logger, indexSettings);
+        addLuceneAvailableCodecs(codecs);
+        this.codecs = codecs.immutableMap();
+    }
+
     public CodecService(@Nullable MapperService mapperService, Logger logger) {
         final MapBuilder<String, Codec> codecs = MapBuilder.<String, Codec>newMapBuilder();
-        if (mapperService == null) {
-            codecs.put(DEFAULT_CODEC, new Lucene95Codec());
-            codecs.put(BEST_COMPRESSION_CODEC, new Lucene95Codec(Mode.BEST_COMPRESSION));
+        if (mapperService != null) {
```

Comment:
👍 but to be 100% safe here, 

```suggestion
        if (mapperService != null && mapperService.getIndexSettings() != null) {
```

Reply:
I think you could still keep the constructors here - otherwise it will be difficult to backport the pull requests, the main & `2.x` would diverge

Reply:
Got it, I've added the two constructors as is